Sources/GRPC/LengthPrefixedMessageWriter.swift

```diff
@@ -44,44 +44,58 @@ internal struct LengthPrefixedMessageWriter {
   /// Writes the data into a `ByteBuffer` as a gRPC length-prefixed message.
   ///
   /// - Parameters:
-  ///   - message: The serialized Protobuf message to write.
+  ///   - payload: The serialized Protobuf message to write.
   ///   - buffer: The buffer to write the message into.
   /// - Returns: A `ByteBuffer` containing a gRPC length-prefixed message.
   /// - Precondition: `compression.supported` is `true`.
   /// - Note: See `LengthPrefixedMessageReader` for more details on the format.
-  func write(_ message: Data, into buffer: inout ByteBuffer, disableCompression: Bool = false) throws {
-    buffer.reserveCapacity(LengthPrefixedMessageWriter.metadataLength + message.count)
-
+  func write(_ payload: GRPCPayload, into buffer: inout ByteBuffer, disableCompression: Bool = false) throws {
+    
+    buffer.reserveCapacity(LengthPrefixedMessageWriter.metadataLength)
+    
     if !disableCompression, let compressor = self.compressor {
       // Set the compression byte.
       buffer.writeInteger(UInt8(1))
-
+      
       // Leave a gap for the length, we'll set it in a moment.
       let payloadSizeIndex = buffer.writerIndex
       buffer.moveWriterIndex(forwardBy: MemoryLayout<UInt32>.size)
 
+      // Build the message by serializing it into a Buffer first
+      // Since deflate can only take a ByteBuffer
```

Comment:
`deflate` can only take a `ByteBuffer` because we decided it should be that way. I don't think we need a comment here to be honest though.